execution_workflows/QAPA/README.md

```diff
@@ -1,11 +1,59 @@
- - replace "path_to" in samplesheet_example_files.csv with the path to the reference files downloaded from the Google Drive link
- - check the path to this directory with `pwd` and replace the `[pwd]` in samplesheet_example_files.csv with the path from the `pwd` command
- - Before running `QAPA`, you have to install `QAPA`, and the guide to install is [here](https://github.com/morrislab/qapa). Make sure that QAPA is on your $PATH
- - Then, you are good to run the pilot benchmark nextflow pipeline with `QAPA`
+The [paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1414-4) is titled QAPA: a new method for the systematic analysis of alternative polyadenylation from RNA-seq data <br>
+The [application download](https://github.com/morrislab/qapa) is free to download
+and [documentation](https://github.com/morrislab/qapa#rna-seq-quantification-of-alternative-polyadenylation-qapa) was used as a reference
+to create the nextflow pipeline flow of this module
+
+## Running the QAPA workflow
+
+### Input & pre-processing
+An example sample sheet is available at `samplesheet_example.csv`. Each row in the samplesheet has nine
+columns:
+
+- sample: name of the sample for logs (e.g control_replicate1)
+- fastq1: FASTQ1 for both single-end and paired-end RNA-seq
+- fastq2: FASTQ2 for paired-end RNA-seq. If single-end, please leave this empty
+
+When using your own data and input file instead of the provided test data and sample sheet, make sure to include in the 
+input file you are using the absolute paths to the files, with the column names following the column
+names above.
+
+### Running with Docker or Singularity
+## Docker
+This workflow uses docker containers. To run with docker, make sure that docker is installed and running 
+(e.g. to ensure docker is running, run the command `docker --help` and a help message should be printed).
+To run with `docker`, please indicate `-profile docker`
+```
+nextflow main.nf --input samplesheet_example_files.csv` --gtf <path_to_gtf> --polyabed <path_to_poly(A)_bed> --fasta <path_to_fasta> --run_qapa_build -profile docker
+```
+
+## Singularity
+To run with `singularity`, please indicate `-profile singularity`
+```
+nextflow main.nf --input samplesheet_example_files.csv` --gtf <path_to_gtf> --polyabed <path_to_poly(A)_bed> --fasta <path_to_fasta> --run_qapa_build -profile singlularity
 ```
-nextflow main.nf --input samplesheet_example_files.csv
+
+### Parameters
+Parameters used to run the two steps of DaPars are specified in conf/modules.config file. 
+Parameters relevant to the workflow itself are:
+- `input` - path to the `samplesheet.csv`
+- `outdir` - name of the folder that the final output files are going to be in, located under QAPA/
+- `gtf`: GTF annotation file
+- `fasta`: FASTA reference sequence file
+- `polyabed`: BED 3'UTR library
+- `run_qapa_build`: run qapa built (default: false)
```

Comment:
So just to be clear our workflow supports two modes:
1. Build annotations from scratch with provided GTF & PolyASite BED file (and pass `--run_qapa_build`)
2. Provide pre-generated QAPA annotations (to `--polyabed`) (and do not pass `--run_qapa_build`/set to false)

Is this correct? I think would be good to add in if so

Reply:
Yes, this is correct. I have added this to the README